patroni/postgresql/sync.py

```diff
@@ -201,35 +221,35 @@ def _handle_synchronous_standby_names_change(self) -> None:
 END;$$""")
         self._postgresql.reset_cluster_info_state(None)  # Reset internal cache to query fresh values
 
-    def current_state(self, cluster: Cluster) -> Tuple[CaseInsensitiveSet, CaseInsensitiveSet]:
-        """Finds best candidates to be the synchronous standbys.
+    def _get_replica_list(self, cluster: Cluster) -> Iterator[Tuple[int, str, str, int, bool]]:
+        """Yields candidates based on higher replay/remote_write/flush lsn.
 
-        Current synchronous standby is always preferred, unless it has disconnected or does not want to be a
-        synchronous standby any longer.
+        .. note::
+            Tuples are reverse ordered by sync_state and LSN fields so nodes that already synchronous or having
+            higher LSN values are preferred.
 
-        Standbys are selected based on values from the global configuration:
-        - `maximum_lag_on_syncnode`: would help swapping unhealthy sync replica in case if it stops
-          responding (or hung). Please set the value high enough so it won't unncessarily swap sync
-          standbys during high loads. Any value less or equal of 0 keeps the behavior backward compatible.
-          Please note that it will not also swap sync standbys in case where all replicas are hung.
-        - `synchronous_node_count`: controlls how many nodes should be set as synchronous.
+        :param cluster: current cluster topology from DCS.
+
+        :yields: tuples composed of:
 
-        :returns: tuple of candidates :class:`CaseInsensitiveSet` and synchronous standbys :class:`CaseInsensitiveSet`.
+            * pid - a PID of walsender process
+            * member name - matches with the application_name
+            * sync_state - one of ("async", "potential", "quorum", "sync")
+            * LSN - write_lsn, flush_lsn, or replica_lsn, depending on the value of ``synchronous_commit`` GUC
+            * nofailover - whether the member has ``nofailover`` tag set
```

Comment:
```suggestion
            * ``pid`` - PID of the walsender process
            * ``member name`` - matches with the ``application_name```
            * ``sync_state`` - one of (``async``, ``potential``, ``quorum``, ``sync``)
            * ``LSN`` - write_lsn, flush_lsn, or replica_lsn, depending on the value of ``synchronous_commit`` GUC
            * ``nofailover`` - whether the member has ``nofailover`` tag set
```

Just highlighting some literals :)